sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/MetricsAdvisorAdministrationClient/NotificationHookLiveTests.cs

```diff
@@ -198,7 +197,7 @@ public async Task UpdateEmailNotificationHookWithMinimumSetupAndGetInstance(bool
 
             hookToUpdate.EmailsToAlert.Add("fake3@email.com");
 
-            await adminClient.UpdateHookAsync(disposableHook.Id, hookToUpdate);
+            await adminClient.UpdateHookAsync(hookToUpdate);
```

Comment:
in theory you don't need to do the get anymore as the update method returns the updated hook, right?

Reply:
same question in other metest methods

Reply:
Not exactly. It's important to understand how Patch models in the REST API work first. For each property of a patch model, there are three possibilities when sending it to the service in an Update call:
- Send the property with the new value you want to attribute to it.
```
{
  "property": "new_value",
   ... // other properties
}
```
- Don't send this property, so the original value will be kept.
```
{
  ... // only other properties
}
```
- Send `null` for this property, so the service will set it to the default value. Some properties (like "name") won't support this operation.
```
{
  "property": null
  ... // other properties
}
```

This is a problem for .NET and Java because we don't have a good way of differentiating between `null` and `undefined`, like JS does.

Originally, in .NET we were trying to let users update a hook/datafeed/config/etc. in two different ways:
1. Make a call to Get to obtain an instance, update the required properties, then send it to the service in an Update method. In this case, no ID parameter is needed because the object already has an `Id` property.
2. Create an empty instance, set only the required properties, and send it to the service in an Update method. In this case, the ID parameter is needed because the `Id` property of the object is initialized with `null`.

Given the complications that arise from implementing such behavior in strongly typed languages, we decided to propose only supporting (1) for this first GA release. We don't have significant customer data for MA, but Johan expects (1) to be the most common scenario even in weakly typed languages.

For this reason, we're removing the ID parameter, because it would only be useful in scenario (2).